parser_library/src/workspace_manager_impl.h

```diff
@@ -203,48 +392,73 @@ class workspace_manager::impl final : public diagnosable_impl
             wks.second.set_message_consumer(consumer);
     }
 
-    void definition(const std::string& document_uri, position pos, workspace_manager_response<position_uri> r)
+    struct
     {
-        location definition_result;
-        auto doc_loc = utils::resource::resource_location(document_uri);
-
-        if (cancel_ && *cancel_)
-        {
-            definition_result = { pos, doc_loc };
-            r.provide(position_uri(definition_result));
-            return;
-        }
+        int code;
+        const char* msg;
+    } static constexpr request_cancelled { -32800, "Canceled" }, request_failed { -32803, "Unknown reason" },
+        removing_workspace { -32803, "Workspace removal in progress" };
```

Comment:
Is it intentional to have -32803 for two different messages?

Reply:
the -32803 means "request failed", the message is just a reason. I included the `request_failed` just to indicate where it came from.